### ISSUE TITLE
SMTP server with alias address sends from credential address instead (server config, not Mail app)

### ISSUE THREAD

**Issue by rafamerino** [nextcloud/server#25341]
When I send email from my alias Google Workspace account, mail is sent from my main username:

Settings in nextcloud server:

Send mode: SMTP
Encryption: STARTTLS
**From address: alias@domain.com <<<<< Here's the address I want to send from**
Auth method: Login (with "Authentication required" checked)
Server address: smtp.gmail.com (port 587)
**Credentials: main.account@domain.com <<<<< Here's the address the system is sending emails**
Password: whatever

### Steps to reproduce
1. Create alias in Google Workspace account
2. Configure Nextcloud (v19 or v20 results in same bug) with SMTP options before
3. Click "Send test email"
4. Email received comes from "main.account@domain.com" instead of "alias@domain.com"

### Expected behaviour
SMTP server configuration does not sent emails with alias saved in "From address" field

### Actual behaviour
Instead, is sending emails with the address in the "Credentials" field

PS: In other systems (i.e. Invoice Ninja), emails are sent correctly from my "alias@domain.com" address)

### Server configuration

**Operating system:**
Debian 9 // Linux 4.9.0-12-amd64 x86_64

**Web server:**
nginx/1.10.3

**Database:**
MariaDB-1:10.4.12

**PHP version:**
7.3.23

**Nextcloud version:**
20.0.6

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from 20.0.5, wich updated from 20.0.4, etc... (bug was there always)

**Where did you install Nextcloud from:**
Source code

Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
No errors have been found.
```

**App list**
<details>
<summary>App list</summary>

**List of activated apps:**
```
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.4
  - bruteforcesettings: 2.0.1
  - calendar: 2.1.3
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contacts: 3.4.3
  - contactsinteraction: 1.1.0
  - dashboard: 7.0.0
  - dav: 1.16.2
  - federatedfilesharing: 1.10.2
  - federation: 1.10.1
  - files: 1.15.0
  - files_downloadactivity: 1.9.0
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.2
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - firstrunwizard: 2.9.0
  - groupfolders: 8.2.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - mail: 1.8.0
  - news: 15.0.2
  - nextcloud_announcements: 1.9.0
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - password_policy: 1.10.1
  - photos: 1.2.3
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - recommendations: 0.8.0
  - richdocuments: 3.7.11
  - richdocumentscode: 6.4.303
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - spreed: 10.0.5
  - support: 1.3.0
  - survey_client: 1.8.0
  - systemtags: 1.10.0
  - tasks: 0.13.6
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - updatenotification: 1.10.0
  - user_status: 1.0.1
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

**Insert your config.php content here.** 
```
<?php
$CONFIG = array (
  'instanceid' => 'xxxxxx',
  'dbtype' => 'mysql',
  'dbhost' => 'localhost',
  'dbname' => 'xxxxxx',
  'dbuser' => 'xxxxxx',
  'dbpassword' => 'xxxxxx',
  'dbtableprefix' => 'xxx_',
  'installed' => true,
  'passwordsalt' => 'xxxxxx',
  'secret' => 'xxxxxx',
  'trusted_domains' =>
  array (
    0 => 'domain.com',
  ),
  'datadirectory' => '/var/www/domain.com/data',
  'version' => '20.0.6.1',
  'overwrite.cli.url' => 'https://domain.com',
  'dbport' => '',
  'mysql.utf8mb4' => true,
  'maintenance' => false,
  'theme' => '',
  'loglevel' => 2,
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'updater.secret' => 'xxxxxx',
  'mail_from_address' => 'alias', <<< alias@domain.com I want to send mails
  'mail_smtpmode' => 'smtp',
  'mail_sendmailmode' => 'smtp',
  'mail_domain' => 'domain.com',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'smtp.gmail.com',
  'mail_smtpport' => '587',
  'mail_smtpname' => 'main@domain.com', <<< main.address@domain.com sending emails
  'mail_smtppassword' => 'xxxxxx',
  'mail_smtpsecure' => 'tls',
  'mail_smtpdebug' => false,
);
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

#### Web server error log
<details>
<summary>Web server error log</summary>

```
(Nginx not showing any error)
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
No errors in nextcloud.log
```
</details>

**Comment by szaimen** [comment]
Is this Issue still valid in NC21.0.2? If not, please close this issue. Thanks! :)

**Comment by rafamerino** [comment]
> Is this Issue still valid in NC21.0.2? If not, please close this issue. Thanks! :)

Yes, same issue in 21.0.2:
Receiver sees "private@domain.com" instead of "contact@domain.com" in "From" field.

![image](https://user-images.githubusercontent.com/371826/123105556-31946d00-d438-11eb-8acd-e0c373ae2733.png)

**Comment by szaimen** [comment]
cc @ChristophWurst

**Comment by ChristophWurst** [comment]
> cc @ChristophWurst

hm?

**Comment by trwnh** [comment]
Still happening in NC 23.0.0

**Comment by trwnh** [comment]
OK, I got this working, I think these are the steps --

1. Go to Gmail settings
2. **Accounts and Import** > **Send mail as** > **Add another email address**
3. Uncheck the "alias" option.
4. Add your "mail from address" to the dialog window and click next.
5. Use your Gmail address (`name@gmail.com`) as the username.
6. Use an app password as the password (go to https://myaccount.google.com/apppasswords to generate one)
7. Save the details

Now, when you send a test email. Gmail should detect the sender as valid. Essentially, it seems that the problem was in Gmail all along and not in Nextcloud.

**Comment by rafamerino** [comment]
That's a workaround, not a real solution.
As I said in my opening post, same config in another systems (i.e. invoice ninja) works flawlessly

Real scenario here is a google workspace email with alias, as I explained in "Steps to reproduce", not the same (I think) as a gmail account.

**Comment by trwnh** [comment]
As best as I can tell, it is Google's SMTP server that refuses to attach the from-address unless there is a send-from address defined that matches the from-address provided by Nextcloud. So it is an issue with Google mail accounts.

**Comment by rafamerino** [comment]
In that case (google problem), neither should work this alias "sent from" in InvoiceNinja or Dolibarr, which works OK and send my emails from my alias address with the _same_ config/parameters as I configured in Nextcloud.

**Comment by gbraad** [comment]
Aliases do not work with a Workspace account? See exactly the same behavior

**Comment by gbraad** [comment]
This NEEDS to be set from Google Mail as follows in the settings: 

![image](https://user-images.githubusercontent.com/1894/212249221-75225445-8b05-4308-b313-71091fcb3aeb.png)

**Comment by gbraad** [comment]
Also, this is an issue that should be in the `/nextcloud/mail` project.

/cc: @szaimen

**Comment by szaimen** [comment]
This looks like a configuration issue then and should be documented in our documentation?

**Comment by rafamerino** [comment]
Thread's original issue is not the case related by @gbraad (**different** domains), so I'm not sure why [this reply](https://github.com/nextcloud/server/issues/25341#issuecomment-1381355412) applies to issue reported by me in opening post (secondary alias but **SAME** domain).

**Comment by szaimen** [comment]
In which NC version did you reproduce the problem?

**Comment by gbraad** [comment]
Sorry if misplaced, but the issue you filed is hard to follow as it actually not very clear what happens. Nonetheless a second alias, even for the same domain, would need to be set first in the Google workspace account before accepted. While other services may work, how NC communicates makes this necessary. Tried with alias for same and other domain.

**Comment by rafamerino** [comment]
Summary from the thread:

> When I send email from my alias Google Workspace account, mail is sent from my main username:
> 
> Settings in nextcloud server:
> 
> Send mode: SMTP
> Encryption: STARTTLS 
> **From address: [alias@domain.com](mailto:alias@domain.com) <<<<< Here's the address I want to send from** 
> Auth method: Login (with "Authentication required" checked) 
> Server address: smtp.gmail.com (port 587) 
> **Credentials: [main.account@domain.com](mailto:main.account@domain.com) <<<<< Here's the address the system is sending emails** 
> Password: whatever
> 
> ### Steps to reproduce
> 1. Create alias in Google Workspace account
> 2. Configure Nextcloud (v19 or v20 results in same bug) with SMTP options before
> 3. Click "Send test email"
> 4. Email received comes from "[main.account@domain.com](mailto:main.account@domain.com)" instead of "[alias@domain.com](mailto:alias@domain.com)"
> 
> ### Expected behaviour
> SMTP server configuration does not sent emails with alias saved in "From address" field
> 
> ### Actual behaviour
> Instead, is sending emails with the address in the "Credentials" field
> 
> PS: In other systems (i.e. Invoice Ninja), emails are sent correctly from my "[alias@domain.com](mailto:alias@domain.com)" address)

> ![image](https://user-images.githubusercontent.com/371826/123105556-31946d00-d438-11eb-8acd-e0c373ae2733.png)

For the "wich NC version" question:
> 2. Configure Nextcloud (v19 or v20 results in same bug)
> Yes, same issue in 21.0.2
> Still happening in NC 23.0.0

And for the free/paid account topic:
> Real scenario here is a google workspace email with alias, as I explained in "Steps to reproduce", not the same (I think) as a gmail account.

**Comment by szaimen** [comment]
Thanks! Is this also still happening with 24.0.9 or 25.0.3?
